### ISSUE TITLE
Raspberry Pi Kernel

### ISSUE THREAD

**Issue by antoniotorres** [xxorde/librekinect#12]
Hi, I don't know much about kernels, but is there some way to have a precompile kernel for the raspberry pi with librekinect already installed? Maybe a modified raspbian image?

Another question, I haven't tested it yet, but if it's working why isn't this shipped with the default raspbian kernels?

**Comment by seanlim1** [comment]
@TimboInSpace I was hoping I could get the Raspbian image from you. I'm having big trouble trying to compile the kernel myself. Many thanks :)

**Comment by TimboInSpace** [comment]
Sure I can send you the current image of my project. It's mostly a blank
wheezy install but has the driver added to the kernel.
We could transfer by ftp if you want
Tim

**Comment by seanlim1** [comment]
Yes please. That would be of  soo much help. How do we go about it?

-Sean

Sent from Windows Mail

**Comment by sjobeek** [comment]
@TimboInSpace I'm also working on a project needing Kinect data streaming over ROS on the network from a Raspi - I'd very much appreciate a copy of your image to build on your work if possible. 

<my_git_name>@gmail.com if you need to send a link privately. 

Thanks!

**Comment by matkinect** [comment]
Hi TimbolnSpace,
I'm working on a project with the Kinect too. 
It would be really helpfull if you could send me a link to download your raspbian distri with the driver added to the kernel.
(my email is in my profile.).

Thanks,
Cheers,
Mat

**Comment by buzztiaan** [comment]
@TimboInSpace i would also like to have your image! :D

**Comment by UserXH726** [comment]
@TimboInSpace i would also like to have your image, too!
adressbedummy@t-online.de

**Comment by tiancovici** [comment]
would really help if you could send an image.
How can I contact you for a copy?

**Comment by TimboInSpace** [comment]
Hi, Sorry everyone I have been listening to your requests:
I've been working on other projects since December, but recently re-started
work on this one. The problem was that, around christmas (but after I had
posted that message) I had gotten my raspberry pi image to a much more
useful state - I wrote all the necessary code and modified my opencv build
to incorporate this librekinect driver into ROS Indigo, but it still had a
few issues within the ROS stuff so I didn't want to post a half-finished
copy.
As of today, I have it so the raspi reads the depth image, formats the
depth image to a 640x40 wide strip, then publishes the image along with
camera_info definitions to be picked up on a faster computer for processing
the depthimage into a pseudo-laserscan. Today, I have it so that's all
working but instead of reading depth into the laserscan, it's giving
inverse-depth - I have a good idea of how to fix it but am working on that
code today.
There's also some issue of a corrupt block somewhere on the SD, but I've
since overwritten the error using dd, and is running fine.

Long story short, I'll post whatever, but it's definitely not finished. It
is however, very very close to having the whole ROS navigation stack
completed.
Is it possible to post whole images on github? I'm not really sure how, but
I'll look into it. Post a reply if you have insight. My current plan is
just to send it personally by ftp, to whomever requests by email - but of
course that's not an ideal solution.

**Comment by tiancovici** [comment]
That'd be great, my email is mokmassadm@gmail.com

**Comment by Kerhold** [comment]
I would gladly have a look as well. kerhold@gmail.com
I'm not sure either if it's possible to put image in github. It's big file,
so might not be possible, otherwise everyone would use it for file sharing,
not code sharing.

2015-03-17 16:23 GMT+01:00 TimboInSpace <notifications@github.com>:

> Hi, Sorry everyone I have been listening to your requests:
> I've been working on other projects since December, but recently re-started
> work on this one. The problem was that, around christmas (but after I had
> posted that message) I had gotten my raspberry pi image to a much more
> useful state - I wrote all the necessary code and modified my opencv build
> to incorporate this librekinect driver into ROS Indigo, but it still had a
> few issues within the ROS stuff so I didn't want to post a half-finished
> copy.
> As of today, I have it so the raspi reads the depth image, formats the
> depth image to a 640x40 wide strip, then publishes the image along with
> camera_info definitions to be picked up on a faster computer for processing
> the depthimage into a pseudo-laserscan. Today, I have it so that's all
> working but instead of reading depth into the laserscan, it's giving
> inverse-depth - I have a good idea of how to fix it but am working on that
> code today.
> There's also some issue of a corrupt block somewhere on the SD, but I've
> since overwritten the error using dd, and is running fine.
>
> Long story short, I'll post whatever, but it's definitely not finished. It
> is however, very very close to having the whole ROS navigation stack
> completed.
> Is it possible to post whole images on github? I'm not really sure how, but
> I'll look into it. Post a reply if you have insight. My current plan is
> just to send it personally by ftp, to whomever requests by email - but of
> course that's not an ideal solution.
>
> On Tue, Mar 17, 2015 at 4:57 AM, tiancovici <notifications@github.com>
> wrote:
>
> > would really help if you could send an image.
> > How can I contact you for a copy?
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/xxorde/librekinect/issues/12#issuecomment-82272826>.
>
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/xxorde/librekinect/issues/12#issuecomment-82409923>.
>



-- 
Pozdrawiam.
K.

**Comment by buzztiaan** [comment]
i'd say just make a account on mega.co.nz and throw it on there

**Comment by tiancovici** [comment]
We can always do Dropbox or Google drive.

**Comment by TimboInSpace** [comment]
Yep, looks like github won't allow repository over 1GB.  Google drive seems
like the best option to me. When I take my next project image,  I'll post
it on Drive and give the link here - expect 1 or 2 days

**Comment by matkinect** [comment]
Hi TimbolnSpace,

I'm not sure if I missed THE post, but could you release the img ?

Cheers,
Mat

**Comment by TimboInSpace** [comment]
Hey all, sorry for the inactivity. In my final year of engineering and it's turning out to be a lot more work than just the courseload!
I'm now uploading to google drive, and I'll post a link here. Looks like it might take a couple hours to upload. It's an old image from mid march, but I believe I havent changed the raspi portion of my project since then. Regardless, when you go to the link, just pick the latest image.
When uploaded, I'll post a rundown of my system and how to operate it.

**Comment by TimboInSpace** [comment]
Here is the link to my project folder, I'm written some release notes and am writing a How-to-run document
https://drive.google.com/open?id=0B5MFGM_aue_afkZnRzljc213NkxTeEtNRFJOSnY3ak9MekZTRThadS1FVDVZMWFPNDhSTEU&authuser=0

**Comment by buzztiaan** [comment]
awesome! thanks a lot!

**Comment by tiancovici** [comment]
Might need to get an 8-16gb disk
but so far when loading it to a 64g card it boots into the rainbow screen. =/

All I did was
1. Erase sd card, and unmount it
2. Flash it via terminal "sudo dd if=ArminMar15-noerrored.img of=/dev/disk2 bs=2m"

**Comment by TimboInSpace** [comment]
Like it stops at the rainbow screen and you never get a prompt? Post this
on my repo under Issues and provide some more detail, I'm sure we can get
it running
